demo.html

```diff
@@ -0,0 +1,262 @@
+                        // c2[0] is color 2
+                        var c2 = blend([0, 255, 0], light)
+
+                        if (ix % 2 == iy % 2)
+                            c.fillStyle = "rgb(" + c1[0] + "," + c1[1] + "," + c1[2] + ")"
+                        else
+                            c.fillStyle = "rgb(" + c2[0] + "," + c2[1] + "," + c2[2] + ")"
+
+                        let project = (x, y, z) => {
+                            away = ((i_hat[1] * x + j_hat[1] * y) - y_Ra[0]) / (y_Ra[1] - y_Ra[0])
+
+                            return [0.25 * away + (-0.5 * away + 1) * ((i_hat[0] * x + j_hat[0] * y) - x_Ra[0]) / (x_Ra[1] - x_Ra[0]), away + z / ((y_Ra[1] - y_Ra[0] + x_Ra[1] - x_Ra[0]) / 2)]
+                        }
+
+                        p1 = project(x, y, evalPts[ix][iy])
+                        p2 = project(x + (x_Ra[1] - x_Ra[0]) / steps, y, evalPts[ix + 1][iy])
+                        p3 = project(x + (x_Ra[1] - x_Ra[0]) / steps, y + (y_Ra[1] - y_Ra[0]) / steps, evalPts[ix + 1][iy + 1])
+                        p4 = project(x, y + (y_Ra[1] - y_Ra[0]) / steps, evalPts[ix][iy + 1])
+
+                        c.beginPath()
+                        var w4 = w / 4
+                        var h4 = h / 4
+                        c.moveTo(w4 * (2 * p1[0] + 1), h4 * (2 * p1[1] + 1))
+                        c.lineTo(w4 * (2 * p2[0] + 1), h4 * (2 * p2[1] + 1))
+                        c.lineTo(w4 * (2 * p3[0] + 1), h4 * (2 * p3[1] + 1))
+                        c.lineTo(w4 * (2 * p4[0] + 1), h4 * (2 * p4[1] + 1))
+                        c.lineTo(w4 * (2 * p1[0] + 1), h4 * (2 * p1[1] + 1))
+                        c.fill()
+
+                        iy++
+                        y += sizes[1]
+
+                    }
+
+                    ix++
+                    x += sizes[0]
+
+                }
+
+            }
+
+            var refresh = () => {
+                // 0.005 represent rotation speed
+                theta += 0.005
+                tanTheta = Math.tan(theta)
+                tanThetaSquared = tanTheta * tanTheta
+                theta %= 6.28318
+
+                var val = Math.sqrt(1 / (tanThetaSquared + 1))
+                var eval = Math.sqrt(-(i_hat[0] * i_hat[0]) + 1)
+                if (tanTheta >= 0) {
+
+                    if (theta > 3.14159) {
+
+                        i_hat[0] = -val
+                        i_hat[1] = -eval
+
+                    } else {
+
+                        i_hat[0] = val
+                        i_hat[1] = eval
+
+                    }
+
+                } else {
+
+                    i_hat[0] = -val
+                    i_hat[1] = eval
+
+                }
+
+                var nTheta = (theta + 1.570795) % 6.28318
+                var tanNTheta = Math.tan(nTheta)
+                var tanNThetaSquared = tanNTheta * tanNTheta
+                var val2 = Math.sqrt(1 / (tanNThetaSquared + 1))
+                var eval2 = Math.sqrt(-(j_hat[0] * j_hat[0]) + 1)
```

Comment:
It's better to use:
```suggestion
                let nTheta = (theta + 1.570795) % 6.28318
                let tanNTheta = Math.tan(nTheta)
                let tanNThetaSquared = tanNTheta * tanNTheta
                let val2 = Math.sqrt(1 / (tanNThetaSquared + 1))
                let eval2 = Math.sqrt(-(j_hat[0] * j_hat[0]) + 1)
```

Or if you are not changing them;
```suggestion
                const nTheta = (theta + 1.570795) % 6.28318
                const tanNTheta = Math.tan(nTheta)
                const tanNThetaSquared = tanNTheta * tanNTheta
                const val2 = Math.sqrt(1 / (tanNThetaSquared + 1))
                const eval2 = Math.sqrt(-(j_hat[0] * j_hat[0]) + 1)
```
Since both `let` and `const` add more security.

Reply:
`const` can't be used here since `theta` changes during the rotation phase. However, `let` would work fine.